mergify_engine/engine/__init__.py

```diff
@@ -205,7 +206,63 @@ def run(client, pull, sub, sources):
         s["event_type"] == "pull_request" and s["data"]["action"] == "closed"
         for s in ctxt.sources
     ):
-        actions_runner.delete_last_summary_head_sha(ctxt)
+        ctxt.clear_cached_last_summary_head_sha()
 
     ctxt.log.debug("engine handle actions")
     actions_runner.handle(mergify_config["pull_request_rules"], ctxt)
+
+
+async def create_initial_summary(owner, event_type, data):
+    if event_type != "pull_request":
+        return
+
+    if data["action"] != "opened":
+        return
+
+    redis = await utils.get_aredis_for_cache()
+    config_location_cache = (
+        f"config-location~{owner}~{data['pull_request']['base']['repo']['name']}"
+    )
```

Comment:
Could we have a method to construct this?
It looks to me it's in 2 different places now.